README.md

```diff
@@ -1,2 +1,52 @@
 # GoalsARM
 HIV transmission dynamic model
+
+## Development
+
+### Prerequisites
+
+* [CMake](https://cmake.org/) (>=3.15) for packaging.
+* [Boost](https://www.boost.org/users/download/) (>=1.82, <=1.85>) installed. The easiest way I find to do this, on windows, is to install one of the prebuilt binaries linked from the download page.
+* [Catch2](https://github.com/catchorg/Catch2) (optional) for testing. It will be fetched by CMake if you do not have it installed locally.
+
+### Build with CMake
+
+```console
+cmake -S . -B build
+cmake --build build
+```
+
+To exclude compiling test code add
+```console
+cmake -S . -B build -DBUILD_TESTING=off
+cmake --build build
+```
+
+Note that when building with testing off nothing is going to be compiled above, this is a header only library and so this will only fetch dependencies and export a target so this can be compiled downstream.
+
+### Testing
+
+```console
+cmake -S . -B build
```

Comment:
I'm having some trouble here. Each time I run this, I get the same message that Catch2 was not found and was fetched dynamically. There's no message indicating that Catch2 was not fetched, but building the tests fails after stepping through this procedure.
```
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22631.
-- Locating Catch2
-- Could NOT find Catch2 (missing: Catch2_DIR)
--   Catch2 not found, fetching dynamically...
-- Configuring done (1.6s)
-- Generating done (0.9s)
-- Build files have been written to: C:/Projects/Repositories/GoalsARM/build
```
I'm a bit stymied, any thoughts on what to try next?

Reply:
So what's happening here is that as it doesn't find Catch2 locally it downloads the library from github. So this build output here looks ok to me. I'll add a message that says when it has fetched successfully to make it a bit clearer.

After you run this, does it fail when you run `cmake --build build`, or `ctest --test-dir build/tests` or do they run ok? What output are you getting from them?

Reply:
Here's a more complete log of what I see:
`PS C:\Projects\Repositories\GoalsARM> cmake -S . -B build
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22631.
-- Locating Catch2
-- Could NOT find Catch2 (missing: Catch2_DIR)
--   Catch2 not found, fetching dynamically...
-- Configuring done (2.1s)
-- Generating done (1.5s)
-- Build files have been written to: C:/Projects/Repositories/GoalsARM/build
PS C:\Projects\Repositories\GoalsARM> cmake --build build
MSBuild version 17.11.9+a69bbaaf5 for .NET Framework

PS C:\Projects\Repositories\GoalsARM> ctest --test-dir build/tests
Internal ctest changing into directory: C:/Projects/Repositories/GoalsARM/build/tests
Test project C:/Projects/Repositories/GoalsARM/build/tests
    Start 1: tests_NOT_BUILT-b12d07c
Could not find executable tests_NOT_BUILT-b12d07c
Looked in the following places:
tests_NOT_BUILT-b12d07c
tests_NOT_BUILT-b12d07c.exe
Release/tests_NOT_BUILT-b12d07c
Release/tests_NOT_BUILT-b12d07c.exe
Debug/tests_NOT_BUILT-b12d07c
Debug/tests_NOT_BUILT-b12d07c.exe
MinSizeRel/tests_NOT_BUILT-b12d07c
MinSizeRel/tests_NOT_BUILT-b12d07c.exe
RelWithDebInfo/tests_NOT_BUILT-b12d07c
RelWithDebInfo/tests_NOT_BUILT-b12d07c.exe
Deployment/tests_NOT_BUILT-b12d07c
Deployment/tests_NOT_BUILT-b12d07c.exe
Development/tests_NOT_BUILT-b12d07c
Development/tests_NOT_BUILT-b12d07c.exe
Unable to find executable: tests_NOT_BUILT-b12d07c
1/1 Test #1: tests_NOT_BUILT-b12d07c ..........***Not Run   0.00 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.02 sec

The following tests FAILED:
          1 - tests_NOT_BUILT-b12d07c (Not Run)
Errors while running CTest
Output from these tests are in: C:/Projects/Repositories/GoalsARM/build/tests/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
PS C:\Projects\Repositories\GoalsARM>`